contrib/database/sql/internal/dsn.go

```diff
@@ -32,7 +33,15 @@ func ParseDSN(driverName, dsn string) (meta map[string]string, err error) {
 			return
 		}
 	default:
-		// not supported
+		// Try to parse the DSN and see if the scheme contains a known driver name.
+		u, err := url.Parse(dsn)
+		if err != nil {
+			// dsn is not a valid URL, so just ignore
+			break
+		}
+		if driverName != u.Scheme {
+			return ParseDSN(u.Scheme, dsn)
+		}
```

Comment:
Can you add a comment explaining this a little bit? (it's not immediately clear why this happens if the `driverName` doesn't match the URL scheme

Reply:
yup!